src/cockpit/superuser.py

```diff
@@ -98,8 +98,22 @@ async def do_custom_command(self, command: str, args: Tuple, fds: List[int], std
 class AuthorizeResponder(CockpitResponder):
     def __init__(self, router: Router):
         self.router = router
+        # in noninteractive init mode (explicit-superuser) we only ever ask ws for the password once
+        # if it doesn't work, give up; after init mode, we ask interactively, so no limitation
+        self.init_mode = True
```

Comment:
I don't think that this is necessary.  We only use the AuthorizeResponder in init mode, so it should always be in init mode.

After that, we use the `SuperuserRoutingRule` itself as the responder, which runs on a different path (see its implementation of `do_askpass()`).

Reply:
Ah, cool! this is confusing, as `SuperuserRoutingRule` instantiates/uses AuthorizeResponder. Cleaned up.